cleverhans/experimental/certification/dual_formulation.py

```diff
@@ -344,14 +455,30 @@ def np_vector_prod_fn_h(np_vector):
                                np.multiply(self.switch_indices[i],
                                            np.maximum(new_lambda_neg_val[i], 0)))
 
-    dual_feed_dict.update(zip(self.lambda_lu, new_lambda_lu_val))
-    dual_feed_dict.update(zip(self.lambda_neg, new_lambda_neg_val))
-    scalar_f = self.sess.run(self.scalar_f, feed_dict=dual_feed_dict)
-    vector_g = self.sess.run(self.vector_g, feed_dict=dual_feed_dict)
-    x, _ = lgmres(linear_operator_h, vector_g)
-    x = x.reshape((x.shape[0], 1))
-    second_term = np.matmul(np.transpose(vector_g), x) + 0.05
+    # Assign new lambda
+    self.sess.run([tf.assign(var, val) for var, val in zip(self.lambda_lu, new_lambda_lu_val)])
```

Comment:
As we discussed before, temporary reassigning variables is not the best way to do it.

Better way to do it is to have a loop inside TensorFlow graph which would iterate through various values.

Please add TODO to resolve it in the future